File: PYTHON/Laptop.py
from PerangkatElektronik import PerangkatElektronik

class Laptop(PerangkatElektronik):
    def __init__(self, merk='', nama='', tahun=0, tipe_keyboard='', baterai=None, ram=None, storage=None, cpu=None, display=None):
        super().__init__(merk, nama, tahun)  # Memanggil constructor dari PerangkatElektronik
        self.tipe_keyboard = tipe_keyboard
        self.baterai = baterai  # Komponen baterai
        self.ram = ram  # Komponen RAM
        self.storage = storage  # Komponen Storage
        self.cpu = cpu  # Komponen CPU
        self.display = display  # Komponen Display

    # Setter & Getter untuk atribut tambahan
    def set_tipe_keyboard(self, tipe_keyboard):
        self.tipe_keyboard = tipe_keyboard

    def get_tipe_keyboard(self):
        return self.tipe_keyboard

    def get_baterai(self):
        return self.baterai

    def get_ram(self):
        return self.ram

    def get_storage(self):
        return self.storage

    def get_cpu(self):
        return self.cpu

    def get_display(self):
        return self.display

File: CPP/Laptop.cpp
#pragma once
#include "Komponen.cpp"
#include "Baterai.cpp"
#include "RAM.cpp"
#include "Storage.cpp"
#include "CPU.cpp"
#include "Display.cpp"
#include "PerangkatElektronik.cpp"

class Laptop : public PerangkatElektronik {
private:
    string tipe_keyboard;
    Baterai baterai;
    RAM ram;
    Storage storage;
    CPU cpu;
    Display display;
    
public:
    // Constructor
    Laptop(string merk, string nama, int tahun, string tipe_keyboard, Baterai b, RAM r, Storage s, CPU c, Display d) {
        this->setMerk(merk);
        this->setNama(nama);
        this->setTahun(tahun);
        this->tipe_keyboard = tipe_keyboard;
        this->baterai = b;
        this->ram = r;
        this->storage = s;
        this->cpu = c;
        this->display = d;
    }

    // Getter & Setter
    void setTipeKeyboard(string tipe_keyboard) {
        this->tipe_keyboard = tipe_keyboard;
    }

    string getTipeKeyboard() {
        return tipe_keyboard;
    }

    Baterai getBaterai() {
        return baterai;
    }

    RAM getRAM() {
        return ram;
    }

    Storage getStorage() {
        return storage;
    }

    CPU getCPU() {
        return cpu;
    }

    Display getDisplay() {
        return display;
    }

    // Destructor
    ~Laptop() {
        // Optional: add any clean up code if needed
    }
};
